src/iaux-underlined-tab-bar.ts

```diff
@@ -0,0 +1,195 @@
+import {
+  html,
+  css,
+  LitElement,
+  property,
+  customElement,
+  query,
+  queryAll,
+  PropertyValues,
+  internalProperty,
+  CSSResult,
+} from 'lit-element';
+import { nothing } from 'lit-html';
+import './iaux-underlined-tab-bar-loading-dots';
+
+export interface TopTab {
+  displayName: string;
+  tabTitle?: string;
+}
+
+@customElement('iaux-underlined-tab-bar')
+export class UnderlinedTabBar extends LitElement {
+  @property({ type: Array }) entries: TopTab[] = [];
+
+  @property({ type: Number }) selectedIndex = 0;
+
+  @property({ type: Number }) widthMultiplier = 0.8;
+
+  @property({ type: Boolean }) isLoading = false;
+
+  @internalProperty() animatedUnderline = false;
+
+  @query('.horizontalScroll') container!: HTMLDivElement;
+
+  @queryAll('button') listOfButtons!: HTMLButtonElement[];
+
+  private itemClicked(e: Event, index: number) {
+    const event = new CustomEvent('itemclicked', {
+      detail: {
+        index,
+      },
+    });
+    const indexDistance = Math.abs(this.selectedIndex - index);
+    const animationTime = Math.min(0.25 * indexDistance, 0.5);
+    this.style.setProperty(
+      '--tabBarUnderlineAnimationDuration',
+      `${animationTime}s`
+    );
+    this.selectedIndex = index;
+    this.dispatchEvent(event);
+    this.animatedUnderline = true;
+  }
+
+  private updateUnderline() {
+    const buttonSelected = this.listOfButtons[this.selectedIndex];
```

Comment:
suggestion: since this const is used below as a button (ie a noun), to my eyes the name `selectedButton` would communicate more directly that the object is a button (and not a boolean state of whether or not the button has been selected)